tracer/src/Datadog.Trace/IAST/Propagation/StringModuleImpl.cs

```diff
@@ -0,0 +1,303 @@
+using Datadog.Trace.Iast.Dataflow;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Iast.Propagation;
+
+internal static class StringModuleImpl
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(StringModuleImpl));
+
+    internal static bool CanBeTainted(string? txt)
+    {
+        return txt != null && txt.Length > 0;
+    }
+
+    internal static TaintedObject? GetTainted(TaintedObjects taintedObjects, object? value)
+    {
+        return value == null ? null : taintedObjects.Get(value);
+    }
+
+    /// <summary> Mostly used overload </summary>
+    /// <param name="left"> Param 1 </param>
+    /// <param name="right"> Param 2</param>
+    /// <param name="result"> Result </param>
+    /// <param name="filter"> Literal filter </param>
+    /// <returns> resiñt </returns>
+    public static string OnStringConcat(string left, string right, string result, AspectFilter filter = AspectFilter.None)
+    {
+        try
+        {
+            if (string.IsNullOrEmpty(left) || string.IsNullOrEmpty(right))
+            {
+                return result;
+            }
+
+            if (!CanBeTainted(result) || (!CanBeTainted(left) && !CanBeTainted(right)))
+            {
+                return result;
+            }
+
+            var iastContext = IastModule.GetIastContext();
+            if (iastContext == null)
+            {
+                return result;
+            }
+
+            TaintedObjects taintedObjects = iastContext.GetTaintedObjects();
+            TaintedObject? taintedLeft = filter != AspectFilter.StringLiteral_1 ? GetTainted(taintedObjects, left) : null;
+            TaintedObject? taintedRight = filter != AspectFilter.StringLiteral_0 ? GetTainted(taintedObjects, right) : null;
+            if (taintedLeft == null && taintedRight == null)
+            {
+                return result;
+            }
+
+            Range[]? ranges;
+            if (taintedRight == null)
+            {
+                ranges = taintedLeft!.Ranges;
+            }
+            else if (taintedLeft == null)
+            {
+                ranges = new Range[taintedRight!.Ranges!.Length];
+                Ranges.CopyShift(taintedRight!.Ranges, ranges, 0, left.Length);
+            }
+            else
+            {
+                ranges = Ranges.MergeRanges(left.Length, taintedLeft!.Ranges, taintedRight!.Ranges);
+            }
+
+            taintedObjects.Taint(result, ranges);
+        }
+        catch (Exception err)
+        {
+            Log.Error(err, "StringModuleImpl.OnstringConcat(string,string) exception {Exception}", err.Message);
+        }
+
+        return result;
+    }
+
+    /// <summary> Overload for multi params (up to 5) </summary>
+    /// <param name="parameters"> StringConcat params struct </param>
+    /// <param name="result"> Result </param>
+    /// <returns> result </returns>
+    public static string OnStringConcat(StringConcatParams parameters, string result)
```

Comment:
nit: use the `in` keyword and pass the struct by ref.

Reply:
You are right. Done!